dapps/shop/src/pages/admin/settings/Console.js

```diff
@@ -1,15 +1,159 @@
 import React, { useState } from 'react'
-import { get } from '@origin/ipfs'
+import ethers from 'ethers'
+
+import { get, post, getIpfsHashFromBytes32 } from '@origin/ipfs'
 
 import useConfig from 'utils/useConfig'
 import useShopConfig from 'utils/useShopConfig'
 
+const abi = [
+  'function updateListing(uint256, bytes32, uint256)',
+  'event ListingCreated (address indexed party, uint indexed listingID, bytes32 ipfsHash)',
+  'event ListingUpdated (address indexed party, uint indexed listingID, bytes32 ipfsHash)'
+]
+
+/**
+ * Gets a marketplace listing's most recent IPFS hash by querying the ethereum network
+ * for relevant logs.
+ * Note: In the future we may want to consider storing the listing's most recent
+ * hash in the shop back-end DB.
+ *
+ * @param {Object} provider: ether.js provider.
+ * @param {Object} contract: ether.js contract object for the marketplace.
+ * @param {Integer} contractEpoch: block number at which contract was created.
+ * @param {string} seller: ethereum address of the seller
+ * @param {Integer} listingId: id of the listing in the marketplace contract.
+ * @returns {Promise<string>} IPFS hash
+ * @private
+ */
+async function _getListingLatestIpfsHash(
```

Comment:
Do we need to bother with knowing the old hash? Can we just overwrite it with fresh listing data, regardless of what it was before?

Reply:
Yes, we could. I was worried that in the future we may store more data in the listing's JSON and that it would get lost if we overwrite it.

But in the short-term we could overwrite with the generic base listing data. I'm happy to put a comment in there so that we remember in the future and to simplify it for the short-term.
Lmk if you have any preference...

Reply:
I like the idea of leaving the schema open and only impacting what this admin cares about.  Though functionally we probably don't need to, it's nice to have.

Reply:
Yea I think we can keep it simple in the short term so we're not requesting logs 😊

Reply:
Ok. I'll leave the _getListingLatestIpfsHash method as a util in the code but for the short-term I won't call it. We can wire it later if we decide to...